### ISSUE TITLE
Richapi.Error with context.workbook.names.add on newly copied sheet

### ISSUE THREAD

**Issue by christian-sap** [OfficeDev/office-js#2875]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

# Provide required information needed to triage your issue
The API call context.workbook.names.add throws an error when you run it on a newly copied sheet

## Your Environment
* Platform [PC desktop, Mac, iOS, Office on the web]: PC desktop
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: all versions, e.g. latest Semi Annual Enterprise
* Operating System: Windows 10
 
## Expected behavior
The API should not throw an error

## Current behavior
The API throws an error

## Steps to reproduce
1. Open the [attached workbook](https://github.com/OfficeDev/office-js/files/9671411/Sample.xlsx) in Windows Excel
2. Run [this ScriptLab snippet](https://github.com/OfficeDev/office-js/files/9671401/copysheet.txt) and click "Run" --> no error
3. Copy Sheet1 -> Sheet1(2) is selected. Do not change the sheet selection
4. Click "Run" again --> error:
"RichApi.Error: A resource with the same name or identifier already exists."

# Provide additional details
If you change sheet selection after step 3 then the error does not come up.

## Useful logs
[ERROR]: Error:
{
    "stack": "RichApi.Error: A resource with the same name or identifier already exists.\n    at new n (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.js:25:241096)\n    at i.processRequestExecutorResponseMessage (https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.js:25:305262)\n    at https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.js:25:303325",
    "message": "A resource with the same name or identifier already exists.",
    "name": "RichApi.Error",
    "code": "ItemAlreadyExists",
    "traceMessages": [],
    "innerError": null,
    "debugInfo": {
        "code": "ItemAlreadyExists",
        "message": "A resource with the same name or identifier already exists.",
        "errorLocation": "NamedItemCollection.add",
        "statement": "var add = names.add(...);",
        "surroundingStatements": [
            "...",
            "var names = workbook.names;",
            "var itemOrNullObject = names.getItemOrNullObject(...);",
            "var range = itemOrNullObject.getRange();",
            "// Instantiate {range}",
            "itemOrNullObject.delete();",
            "// >>>>>",
            "var add = names.add(...);",
            "// <<<<<"
        ],
        "fullStatements": [
            "Please enable config.extendedErrorLogging to see full statements."
        ]
    },
    "httpStatusCode": 400
}
Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by gingerjia** [comment]
@christian-sap, thanks very much for reported the issue. I can repro locally and reported a bug [BUG 6816893] to track the issue. Could you let us know the impact and we can priority it accordingly? Thanks very much!

**Comment by christian-sap** [comment]
Hi @gingerjia , every issue is annoying. Our ~3000 active end users (per month) potentially run into this issue and get a toaster showing the error in our task pane. Of course this is no severe issue because there is workaround and nothing gets broken but it is the sum of little issues that lead to a bad user experience. So we would be happy if you could resolve this even though it is not prio very high :-)

**Comment by gingerjia** [comment]
@christian-sap, thanks very much for the detail info! I have updated it to the bug also. Will keep you update if have progress. Thanks again for reporting the issue!
